launcher/minecraft/PackProfile.cpp

```diff
@@ -939,6 +942,65 @@ bool PackProfile::installCustomJar_internal(QString filepath)
     return true;
 }
 
+bool PackProfile::installAgents_internal(QStringList filepaths)
+{
+    // FIXME code duplication
+    const QString patchDir = FS::PathCombine(d->m_instance->instanceRoot(), "patches");
+    if (!FS::ensureFolderPathExists(patchDir))
+        return false;
+
+    const QString libDir = d->m_instance->getLocalLibraryPath();
+    if (!FS::ensureFolderPathExists(libDir))
+        return false;
+
+    for (const QString& source : filepaths) {
+        const QFileInfo sourceInfo(source);
+        const QString id = QUuid::createUuid().toString(QUuid::WithoutBraces);
+        const QString targetBaseName = id + ".jar";
+        const QString targetId = "org.prismlauncher.agent." + id;
```

Comment:
I think we should stop using the namespace of the launcher. Maybe just do this:

```suggestion
        const QString targetId = "custom.agent." + id;
```